forge-kernel/src/DirectModeling.cpp

```
@@ -248,6 +248,83 @@ TopoDS_Shape extrudeFace(const TopoDS_Face& face, const NVec3& vec) {
+// Simpson + Newton; this one places points ADAPTIVELY by recursive midpoint
+// subdivision so the chordal polyline tracks curvature — used for viewport edge
+// picking (edgeSegments). Both use ONLY the adaptor's D0/D1 evaluators (TKG3d), so
+// no TKGeomBase symbol is referenced.
+//
+// adaptiveSubdivide: recursively bisect [ta,tb], appending the interior split
+// parameters IN ORDER to `out`. A cell is split while EITHER the sagitta (distance
+// of the true curve midpoint to the chord Pa->Pb) exceeds the curvature tolerance
+// `curvTol`, OR the tangent turn (angle between C'(ta) and C'(tb)) exceeds the
+// angular tolerance `angTol` — the exact two GCPnts_TangentialDeflection stop
+// criteria. Bounded by `maxDepth` and a minimum parameter span `minStep`.
+void adaptiveSubdivide(BRepAdaptor_Curve& ad, double ta, double tb,
```

Clang (14.0.6)

[warning] parameter name 'ad' is too short, expected at least 3 characters

(readability-identifier-length)